scio-core/src/main/scala/com/spotify/scio/coders/instances/JavaCoders.scala

```diff
@@ -69,6 +68,29 @@ trait JavaCoders extends CoderGrammar with JavaBeanCoders {
   implicit def jArrayListCoder[T](implicit c: Coder[T]): Coder[java.util.ArrayList[T]] =
     xmap(jListCoder[T])(new java.util.ArrayList(_), identity)
 
+  /**
+   * Coder must be created explicitly via
+   * {{{implicit val pqCoder = Coders.jPriorityQueueCoder[T](ord)}}} since there is otherwise no
+   * guarantee that the PriorityQueue comparator is the same as an implicitly-available ordering at
+   * coder construction time.
+   *
+   * @param ord
+   *   Ordering used as PriorityQueue comparator
+   */
+  def jPriorityQueueCoder[T: Coder: ClassTag](
+    ord: Ordering[T]
+  ): Coder[java.util.PriorityQueue[T]] = {
+    // neither arrays nor PriorityQueues are consistentWithEquals
+    Coder.xmap(ScalaCoders.arrayCoder[T])(
```

Comment:
I think it would be better to create a coder instance, explicitly overriding `consistentWithEquals` instead of depending on a specific type for that reason